src/EditorFeatures/CSharpTest/ConvertCast/ConvertTryCastToDirectCastTests.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.ConvertCast;
+using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.ConvertConversionOperators
+{
+    using VerifyCS = CSharpCodeRefactoringVerifier<CSharpConvertTryCastToDirectCastCodeRefactoringProvider>;
+
+    [Trait(Traits.Feature, Traits.Features.ConvertCast)]
+    public class ConvertTryCastToDirectCastTests
+    {
+        [Fact]
+        public async Task ConvertFromAsToExplicit()
+        {
+            const string InitialMarkup = @"
+class Program
+{
+    public static void Main()
+    {
+        var x = 1 as[||] object;
+    }
+}";
+            const string ExpectedMarkup = @"
+class Program
+{
+    public static void Main()
+    {
+        var x = (object)1;
+    }
+}";
+            await new VerifyCS.Test
+            {
+                TestCode = InitialMarkup,
+                FixedCode = ExpectedMarkup,
+                CodeActionValidationMode = CodeActionValidationMode.Full,
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task ConvertFromAsToExplicit_ValueType()
+        {
+            const string InitialMarkup = @"
+class Program
+{
+    public static void Main()
+    {
+        var x = 1 as[||] byte;
+    }
+}";
+            await new VerifyCS.Test
+            {
+                TestCode = InitialMarkup,
+                CompilerDiagnostics = CompilerDiagnostics.None, // CS0077 is present, but we present the refactoring anyway (this may overlap with a diagnostic fixer)
```

Comment:
Should test for `ExpectedDiagnostics` instead.

This was another troubling test case I stumbled upon:

* The refactoring was acting like a CodeFix: There was a diagnostic and after applying the code action it was gone
* I was not able to cover this with `ExpectedDiagnostics` because the verifier tried to look up the diagnostic marker in the fixed source, even though I provided different `ExpectedDiagnostics` for TestState and FixedSate.

Maybe I missed something, but I gave up quickly and used `CompilerDiagnostics.None` as a recourse.